### PR TITLE
Port completness react query

### PR DESCRIPTION
**Pull request by olethanh**

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/completeness/index.js [hunk 1]
```diff
@@ -1,76 +1,49 @@
...
-    }
-}
+import { getRequest } from '../../libs/Api';
+import { enqueueSnackbar } from '../../redux/snackBarsReducer';
+import { errorSnackBar } from '../../constants/snackBars';
+
+const Completeness = ({ params }) => {
+    const { formatMessage } = useSafeIntl();
+    const dispatch = useDispatch();
+    const { data = [], isFetching } = useQuery(['completeness'], () =>
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/completeness/index.js:17]: i would use a requests.js file in each domain to group them. We have only one request for now but it's to be consistent/

**Comment by beygorghor** [comment]
Also action column is sortable and sometimes looks useless
Screenshot 2021-09-06 at 13 43 25

**Comment by olethanh** [comment]
Rebased on master, added back test that was removed by error. 
Modified to use useSnackMutation (and move it from polio to iaso), added a useSnackQuery. Both of theses function are just like the regulard useMutation and useQuery but handle the snackbar directly for us. which make the code a bit more readable

File: hat/assets/js/apps/Iaso/domains/completeness/components/CompletenessListComponent.js [hunk 2]
```diff
@@ -11,12 +11,7 @@ import { groupCompletenessData } from '../utils';
...
 const styles = theme => commonStyles(theme);
 
-function CompletenessListComponent({
-    classes,
-    completenessList,
-    redirectTo,
-    onGenerateDerivedInstances,
-}) {
+function CompletenessListComponent({ classes, completenessList }) {
     const [activePeriodType, setActivePeriodType] =
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/completeness/components/CompletenessListComponent.js:15]: small nitpick: I would change to an arrow function as it seems to be the standard way to make function components

File: hat/assets/js/apps/Iaso/domains/completeness/components/CompletenessPeriodComponent.js [hunk 3]
```diff
@@ -47,38 +57,49 @@ const CompletenessPeriodComponent = ({
...
     };
 
     const onClick = form => {
         const periods = Array.from(
             new Set(Object.values(form.months).map(m => m.period.periodString)),
         );
         const derived = form.generate_derived;
-        onGenerateDerivedInstances({ periods, derived });
+        mutation.mutate({ periods, derived });
     };
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/completeness/components/CompletenessPeriodComponent.js:89]: I would rename `mutation.mutate` to make it more expressive

**Review by quang-le — COMMENTED** [review]
It sure trims a lot of fat

**Review by quang-le — APPROVED** [review]

**Pull request merged**